i6_models/parts/conformer/mhsa.py

```diff
@@ -39,17 +39,21 @@ def __init__(self, cfg: ConformerMHSAV1Config):
         )
         self.dropout = cfg.dropout
 
-    def forward(self, input_tensor: torch.Tensor, key_padding_mask: Optional[torch.Tensor] = None) -> torch.Tensor:
+    def forward(self, input_tensor: torch.Tensor, sequence_mask: Optional[torch.Tensor] = None) -> torch.Tensor:
         """
         Apply layer norm and multi-head self attention and dropout
-        :param Optional[torch.Tensor] key_padding_mask: could be a binary or float mask of shape (B, T)
+
+        :param sequence_mask: could be a binary or float mask of shape (B, T), 1 signals within sequence, 0 outside, will be inverted to match the torch.nn.MultiheadAttention module
```

Comment:
First, I would say bool mask, not binary mask.

Then: Do we really want to allow both cases here (bool or float)? We could also just say that we expect a bool mask in any case.

Then, what was the result of the discussion now w.r.t. what we actually want to have? Bool or float? Seq mask or seq len or some dataclass? Inv seq mask or seq mask? On device or CPU? I don't see that in the meeting notes. Or when I left the meeting, I think it was bool mask on GPU, but there was still ongoing discussion, maybe the dataclass was still a valid option?

Reply:
Good catch on the doc, I missed that. Regarding the discussion I cannot say much so maybe @christophmluscher @SimBe195 or someone else could comment.